.github/workflows/test-in-app-messaging-prs.yml

```diff
@@ -0,0 +1,63 @@
+# Description: This workflow runs unit + e2e tests on PRs targeting
+#   - `in-app-messaging/main`
+#   - `in-app-messaging/release`
+#
+# Triggered by:
+#   (1) Internal PRs: maintainer pushes a commit to PRs targeting those branches.
+#   (2) Fork PRs: maintainer adds "run-test" label to PRs targeting those branches.
+
+name: Test / in-app-messaging / PRs
+
+concurrency:
+  group: e2e-${{ github.event.pull_request.id }}
+  cancel-in-progress: true
+
+on:
+  pull_request_target:
+    branches: [
+        in-app-messaging/main
+        in-app-messaging/release,
+      ]
+    types: [opened, synchronize, labeled]
+
+permissions:
+  pull-requests: write # used to remove label
+  # other permissions are defaulted to "none"
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    # We run tests only if it's:
+    #   1) pull request not from a fork (ie. internal PR), or
+    #   2) pull request from a fork (ie. external PR) that was added "run-tests" label
+    if: |
+      (github.event.pull_request.head.repo.full_name == github.repository) ||
+      (github.event.action == 'labeled' && github.event.label.name == 'run-tests')
+    steps:
+      - name: Remove run-tests label, if applicable
+        if: always() && github.event.label.name == 'run-tests'
+        uses: actions/github-script@0.3.0
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const { issue: { number: issue_number }, repo: { owner, repo } } = context;
+            const label = 'run-tests';
+            github.issues.removeLabel({ owner, repo, issue_number, name: label });
+  test:
+    uses: aws-amplify/amplify-ui/.github/workflows/reusable-e2e.yml@in-app-messaging/main
```

Comment:
This is really the only change outside copy+paste. We're changing the reference:

```diff
- uses: aws-amplify/amplify-ui/.github/workflows/reusable-e2e.yml@main
+ uses: aws-amplify/amplify-ui/.github/workflows/reusable-e2e.yml@in-app-messaging/main
```

Reply:
It references this file if you're curious: https://github.com/aws-amplify/amplify-ui/blob/in-app-messaging/main/.github/workflows/reusable-e2e.yml